### PR TITLE
ignore + inactivate MessageThreads with inactive keepIds

### PR DESCRIPTION
**Pull request by camhashemi**
@ryanpbrewster @leogrim @andrewconner 

tested it in dev mode, performance + pagination looks good, but some old and inactive keepIds were breaking things.

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/eliza/ElizaServiceClient.scala [hunk 1]
```diff
@@ -306,8 +306,9 @@ class ElizaServiceClientImpl @Inject() (
   }
 
   def getElizaKeepStream(userId: Id[User], limit: Int, beforeId: Option[Id[Keep]], filter: ElizaFeedFilter): Future[Seq[(Id[Keep], DateTime)]] = {
+    import GetElizaKeepStream._
     call(Eliza.internal.getElizaKeepStream(userId, limit, beforeId, filter)).map { response =>
-      (response.json \ "keepIdsAndLastUpdated").as[Seq[(Id[Keep], DateTime)]]
+      response.json.as[Response].keepIdsAndLastUpdated
```
Comment (ryanpbrewster) [inline kifi-backend/modules/common/app/com/keepit/eliza/ElizaServiceClient.scala]: :+1:

File: kifi-backend/modules/common/app/com/keepit/eliza/ElizaServiceClient.scala [hunk 2]
```diff
@@ -410,6 +411,10 @@ object ElizaServiceClient {
     implicit val requestFormat: Format[Request] = Json.format[Request]
     implicit val responseFormat: Format[Response] = Json.format[Response]
   }
+  object GetElizaKeepStream {
+    case class Response(keepIdsAndLastUpdated: Seq[(Id[Keep], DateTime)])
```
Comment (ryanpbrewster) [inline kifi-backend/modules/common/app/com/keepit/eliza/ElizaServiceClient.scala]: Do you think a `Map[Id[Keep], DateTime]` makes sense here?

File: kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaDiscussionCommander.scala [hunk 3]
```diff
@@ -239,6 +240,7 @@ class ElizaDiscussionCommanderImpl @Inject() (
         }
       }
       uts.foreach(userThreadRepo.deactivate)
+      nuts.foreach(nonUserThreadRepo.deactivate)
```
Comment (ryanpbrewster) [inline kifi-backend/modules/eliza/app/com/keepit/eliza/commanders/ElizaDiscussionCommander.scala:243]: :+1:

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala [hunk 4]
```diff
@@ -845,7 +845,12 @@ class KeepCommanderImpl @Inject() (
         val beforeId = beforeExtId.flatMap(extId => db.readOnlyReplica(implicit s => keepRepo.get(extId).id))
         eliza.getElizaKeepStream(userId, limit, beforeId, filter).map { keepIdsAndLastActivity =>
           val keepsByIds = db.readOnlyReplica(implicit s => keepRepo.getByIds(keepIdsAndLastActivity.map(_._1).toSet))
-          keepIdsAndLastActivity.map { case (keepId, lastActivity) => (keepsByIds(keepId), lastActivity) }
+          Future {
+            // make sure eliza is up-to-date on keeps that have been deleted
+            val zombieThreadKeepIds = keepIdsAndLastActivity.collect { case (keepId, _) if !keepsByIds.contains(keepId) => keepId }
+            eliza.deleteThreadsForKeeps(zombieThreadKeepIds.toSet)
+          }
+          keepIdsAndLastActivity.collect { case (keepId, lastActivity) if keepsByIds.contains(keepId) => (keepsByIds(keepId), lastActivity) }
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala]: This doesn't seem like the place to be doing this (this being deleting message threads).

Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala]: IMO what should happen here is a `keepsByIds.flatMap { case (kId, keep) => keep -> lastActivityByKeepId(kId) }`

File: kifi-backend/modules/common/app/com/keepit/eliza/ElizaServiceClient.scala (hunk 2 above)
Comment (camhashemi) [inline kifi-backend/modules/common/app/com/keepit/eliza/ElizaServiceClient.scala]: I thought about it, not sure what it buys us much other than elegance, and we'd have to resort.

Comment (camhashemi) [inline kifi-backend/modules/common/app/com/keepit/eliza/ElizaServiceClient.scala]: the limit is always 10, so nbd, that's just what I was thinking.

File: kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala [hunk 5]
```diff
@@ -845,7 +845,12 @@ class KeepCommanderImpl @Inject() (
         val beforeId = beforeExtId.flatMap(extId => db.readOnlyReplica(implicit s => keepRepo.get(extId).id))
         eliza.getElizaKeepStream(userId, limit, beforeId, filter).map { keepIdsAndLastActivity =>
           val keepsByIds = db.readOnlyReplica(implicit s => keepRepo.getByIds(keepIdsAndLastActivity.map(_._1).toSet))
-          keepIdsAndLastActivity.map { case (keepId, lastActivity) => (keepsByIds(keepId), lastActivity) }
+          Future {
+            // make sure eliza is up-to-date on keeps that have been deleted
+            val zombieThreadKeepIds = keepIdsAndLastActivity.collect { case (keepId, _) if !keepsByIds.contains(keepId) => keepId }
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala]: Here you can probably do `val zombieKeepIds = lastActivityByKeepId.keySet -- keepsByIds.keySet`

Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/commanders/KeepCommander.scala]: ballin, thank you

**Pull request merged**